.github/workflows/lp-upload.yaml

```
@@ -0,0 +1,47 @@
+on:
+  workflow_call:
+    inputs:
+      ppa:
+        type: string
+        description: PPA of the LP user to upload package
+        default: ppa
+      package:
+        type: string
+        description: Name of package to be uploaded
+        required: true
+
+jobs:
+  launchpad-upload:
+    name: LP Upload
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Configure Dput
+        run: |
+          cp config/.dput.cf ~/
+          sed -i "s/<your_launchpad_id>/${{ secrets.LP_USER }}/g" ~/.dput.cf
+          sed -i "s/<ppa_name>/${{ github.event.inputs.ppa }}/g" ~/.dput.cf
+      - name: Import Keys
+        run: |
+          sudo apt update
+          sudo apt install -y ca-certificates
+          echo "${{ secrets.GPG_PUB_KEY }}" > pub.key
+          echo "${{ secrets.GPG_PRIVATE_KEY }}" > private.key
+          gpg --import pub.key
+          gpg --import private.key
+          rm *.key
+      - uses: actions/download-artifact@v3
+        with:
+          name: ${{ github.event.inputs.package }}
+      - name: Sign Changes
+        run: |
+          sudo apt update
+          sudo apt install -y devscripts
+          FINGERPRINT=$(gpg --list-keys --with-colons --fingerprint | awk -F: '$1 == "fpr" {print $10;}' | awk 'NR==1{print $1}')
+          debsign -k $FINGERPRINT ./build-area/*.changes
+      - name: Upload to Launchpad
+        run: |
+          dput my-ppa ./build-area/*source.changes
```

I'm not sure on the syntax, but maybe this works?

```suggestion
          dput my-ppa ./build-area/${{ github.event.inputs.package }}*source.changes
```

It feels to me like it's helpful to be more specific in case there are multiple projects in the directory. Or maybe since these are CI runners, that would never happen?

---

that shouldnt happen, but its not bad practice to do that.

---

although, looking at it, I may remove the input for the package name altogether as the artifact uploading/downloading is tied to a workflow run and doesnt need to be named uniquely